pkg/cvo/cvo.go

```diff
@@ -591,18 +591,12 @@ func (optr *Operator) sync(ctx context.Context, key string) error {
 	config := validation.ClearInvalidFields(original, errs)
 
 	// identify the desired next version
-	desired, ok := findUpdateFromConfig(config)
-	if ok {
-		klog.V(2).Infof("Desired version from spec is %#v", desired)
-	} else {
-		currentVersion := optr.currentVersion()
-		desired = configv1.Update{
-			Version: currentVersion.Version,
-			Image:   currentVersion.Image,
-		}
-		klog.V(2).Infof("Desired version from operator is %#v", desired)
-	}
+	desired, findUpdateWarning := findDesiredUpdate(config, optr.currentVersion())
+	if findUpdateWarning != nil {
+		klog.Warningf("Deprecated desired update matching: %v", findUpdateWarning)
```

Comment:
The message here is confusing to me - I'm not sure if I understood what is wrong if I saw that in the log. What is deprecated? Is the problem that something is deprecated, or is the problem that something is matching... what?

Also - is there a good reason why not to simply call the second argument `err` (or `findUpdateErr` or whatever)? It's an `error`, and because of the convention is so prevalent, naming it otherwise indicates it's not an error. If I only read callsite code I'd probably think it's a string.

Also 2) It's also a convention that callers should not rely on actual return values if the method returned a non-nil error. I think that the intent that it is fine in this case should be documented (probably on both sides, but at least in `findDesiredUpdate` godoc - if the contract is "no matter the error, this method always retuns something sane in its first return value", then IMO it should be made explicit.

Reply:
The full message would look like:

    Deprecated desired update matching: release version "4.13.0" does not match desired update "4.13.3"

for someone who set:

```yaml
desiredUpdate:
  image: ${PULLSPEC_FOR_4_13_3}
  version: 4.13.0
```

> I think that the intent that it is fine in this case should be documented...

Done with 96dfce7c -> 6c2d7e79